test/runtests.jl

```diff
@@ -164,160 +164,195 @@ end
 
-    gm11 = fit(GeneralizedLinearModel, X, Y, Binomial())
-    @test isapprox(predict(gm11), Y)
+gm11 = fit(GeneralizedLinearModel, X, Y, Binomial())
+@test isapprox(predict(gm11), Y)
 
-    newX = rand(5, 2)
-    newY = logistic.(newX * coef(gm11))
-    @test isapprox(predict(gm11, newX), newY)
+newX = rand(5, 2)
+newY = logistic.(newX * coef(gm11))
+@test isapprox(predict(gm11, newX), newY)
 
-    off = rand(10)
-    newoff = rand(5)
+off = rand(10)
+newoff = rand(5)
 
-    @test_throws ArgumentError predict(gm11, newX, offset=newoff)
+@test_throws ArgumentError predict(gm11, newX, offset=newoff)
 
-    gm12 = fit(GeneralizedLinearModel, X, Y, Binomial(), offset=off)
-    @test_throws ArgumentError predict(gm12, newX)
-    @test isapprox(predict(gm12, newX, offset=newoff),
-        logistic.(newX * coef(gm12) .+ newoff))
+gm12 = fit(GeneralizedLinearModel, X, Y, Binomial(), offset=off)
+@test_throws ArgumentError predict(gm12, newX)
+@test isapprox(predict(gm12, newX, offset=newoff),
+logistic.(newX * coef(gm12) .+ newoff))
 
-        # Prediction from DataFrames
-    d = convert(DataFrame, X)
-    d[:y] = Y
+# Prediction from DataFrames
+d = convert(DataFrame, X)
+d[:y] = Y
 
-    gm13 = fit(GeneralizedLinearModel, y ~ 0 + x1 + x2, d, Binomial())
-    @test predict(gm13) == predict(gm13, d[[:x1, :x2]])
-    @test predict(gm13) == predict(gm13, d)
+gm13 = fit(GeneralizedLinearModel, y ~ 0 + x1 + x2, d, Binomial())
+@test predict(gm13) == predict(gm13, d[[:x1, :x2]])
+@test predict(gm13) == predict(gm13, d)
 
-    newd = convert(DataFrame, newX)
-    predict(gm13, newd)
+newd = convert(DataFrame, newX)
+predict(gm13, newd)
 end
 
 @testset "Issue 118" begin
-    @inferred nobs(lm(randn(10, 2), randn(10)))
+@inferred nobs(lm(randn(10, 2), randn(10)))
 end
 
 @testset "Issue 84" begin
-    X = [1 1; 2 4; 3 9]
-    Xf = [1 1; 2 4; 3 9.]
-    y = [2, 6, 12]
-    yf = [2, 6, 12.]
-    @test isapprox(lm(X, y).pp.beta0, ones(2))
-    @test isapprox(lm(Xf, y).pp.beta0, ones(2))
-    @test isapprox(lm(X, yf).pp.beta0, ones(2))
+X = [1 1; 2 4; 3 9]
+Xf = [1 1; 2 4; 3 9.]
+y = [2, 6, 12]
+yf = [2, 6, 12.]
+@test isapprox(lm(X, y).pp.beta0, ones(2))
+@test isapprox(lm(Xf, y).pp.beta0, ones(2))
+@test isapprox(lm(X, yf).pp.beta0, ones(2))
+end
+
+@testset "F test for model comparison" begin
+    d = DataFrame(Treatment=[1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2.],
+                  Result=[1.1, 1.2, 1, 2.2, 1.9, 2, .9, 1, 1, 2.2, 2, 2],
+                  Other=[1, 1, 2, 1, 2, 1, 3, 1, 1, 2, 2, 1])
+    mod = lm(@formula(Result~Treatment), d).model
+    othermod = lm(@formula(Result~Other), d).model
+    nullmod = lm(@formula(Result~1), d).model
+    @test GLM.issubmodel(nullmod, mod)
+    @test !GLM.issubmodel(othermod, mod)
+    
+    @test_throws ArgumentError ftest(mod, othermod)
+    
+    ft = ftest(mod, nullmod)
+    @test isapprox(ft.cols[end][2],  2.481215056713184e-8)
+    # Test output
+    @test sprint(show, ftest(mod, nullmod)) == 
+"""
```

Comment:
Sorry to be a pain, but this line should have four spaces of indentation more than the last line. Triple quoted strings do not take the indentation into account.